docs/contributor/arch/016a-strategy-for-supported-ottl-functions.md

```diff
@@ -0,0 +1,38 @@
+# 16a. Strategy for Supported OTTL Functions
+
+Date: 25.7.2025
+
+## Status
+
+Open
+
+## Context
+
+OTTL functions enable users to customize their telemetry processing pipelines. While this extensibility brings significant benefits, it also introduces challenges related to performance, stability, and the lifecycle management of supported functions.
+
+## Challenges
+
+- **Stability:**  
+  We currently lack experience with OTTL functions in a production scenario, so their stability is not yet proven.
+
+- **Performance:**  
+  The performance impact of user-defined OTTL configurations is unknown. Certain functions or combinations may have an considerable impact on performance.
+
+- **Deprecations and Removals:**  
+  With every release of otel-collector functions may be deprecated or removed. While deprecated functions may continue to be used we need to guide our users for the upcoming removal of those functions.
+  Once a function is fully removed pipelines still using this function will become invalid and stop working.
+
+
+## Decisions
+
+1. **No Explicit Support List:**  
+   Initially, we considered limiting support to a minimal subset of OTTL functions. This idea was discarded as it will lead to confusion if users refer to public documentation about OTTL.
```

Comment:
```suggestion
   Initially, we considered limiting support to a minimal subset of OTTL functions. This idea was discarded because it will lead to confusion if users refer to public documentation about OTTL.
```